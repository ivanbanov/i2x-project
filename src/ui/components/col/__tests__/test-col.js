import React from 'react';
import renderer from 'react-test-renderer';
import Col from '../';

jest.mock('src/ui/styles/mixins/media-query.styl', () => ({}));
jest.mock('src/ui/styles/setup.styl', () => ({}));
jest.mock('../styles.styl', () => ({ COLS: 12 }));

describe('Col', () => {
  it('should render properly', () => {
    const component1 = renderer.create(
      <Col
        gutter="large"
        size={12}
        className="col-foo-bar"
        style={{ backgroundColor: 'red' }}
      >
        <div>Col</div>
      </Col>
    );

    const component2 = renderer.create(
      <Col
        gutter={{ top: 'large' }}
        size={{ small: 12, large: 6 }}
      />
    );

    const tree = component1.toJSON();
    const tree2 = component2.toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
  });
});
