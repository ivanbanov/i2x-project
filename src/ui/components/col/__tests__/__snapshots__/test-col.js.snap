// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should render properly 1`] = `
<div
  className="undefined undefined undefined undefined undefined col-foo-bar"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <div>
    Col
  </div>
</div>
`;

exports[`Col should render properly 2`] = `
<div
  className="undefined"
  style={undefined}
/>
`;
